File: JavaScript/3147.js
const l = require('fs').readFileSync('/dev/stdin', 'utf8').split('\n').shift().split(" ").map(Number);

let g = 0;
let e = 0;

l.forEach((item, i) => {
    if (i > 2 && i < 5) {
        e += item;
    } else {
        g += item;
    }
});

console.log(g > e ? "Middle-earth is safe." : "Sauron has returned.");

File: Python/3147.py
g = 0
e = 0

l = input().split()
for i in range(len(l)):
    if 2 < i < 5:
        e += int(l[i])
    else:
        g += int(l[i])

print("Middle-earth is safe." if g > e else "Sauron has returned.")